src/components/NavigationMailbox.vue

```diff
@@ -92,7 +92,7 @@
 			</ActionButton>
 
 			<ActionCheckbox
-				v-if="notVirtual"
+				v-if="notVirtual && filter !== 'starred' "
```

Comment:
you could also check if `filter` is not undefined. Then this will still work for any other kind of filtered mailbox.

Reply:
and you could move the logic into `notVirtual`

The part ``this.mailbox.specialRole !== 'flagged'`` seems to actually target the case you are fixing here. Did we lose the specialRole=flagged on the favorites mailbox?

Reply:
> you could also check if filter is not undefined. Then this will still work for any other kind of filtered mailbox.

unfortunately it is not working this way

> and you could move the logic into notVirtual

done

> Did we lose the specialRole=flagged on the favorites mailbox?

i think not, or what do you mean? 

new pushed version works

Reply:
See https://github.com/nextcloud/mail/pull/6411#pullrequestreview-974926360